app/src/main/java/com/example/superimagesviewer/MosaicRepository.java

```diff
@@ -0,0 +1,69 @@
+package com.example.superimagesviewer;
+
+import android.app.Application;
+import android.content.res.AssetManager;
+import android.graphics.drawable.Drawable;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MosaicRepository {
+
+    private static final String IMAGES_FOLDER_NAME = "MyImages";
+
+    private MutableLiveData<List<Drawable>> mosaicsList = new MutableLiveData<>();
+    private final List<Drawable> drawables = new ArrayList<>();
+
+    MosaicRepository(Application application) {
+        new InsertAsyncTask(application).execute();
+    }
+
+    public LiveData<List<Drawable>> getMosaicsList() {
+        return mosaicsList;
+    }
+
+    private class InsertAsyncTask extends AsyncTask<Void, Void, Void> {
+
+        private final Application application;
+
+        public InsertAsyncTask(Application application) {
+            this.application = application;
```

Comment:
This could a potential memory leak, please search for "Weak reference". Also I recommend to search for references types

Reply:
done